routes/users.js

```diff
@@ -3,19 +3,34 @@ const express = require('express');
 const router = express.Router();
 
 let { User } = require('../models/user.model');
+let admin = require('firebase-admin');
+
+admin.initializeApp({
+	credential: admin.credential.applicationDefault(),
+	databaseURL: 'https://bruinbot-8d68e.firebaseio.com',
+});
 
 router.route('/').get((req, res) => {
 	User.find()
 		.then((users) => res.json(users))
 		.catch((err) => res.status(400).json(err));
 });
 
-router.route('/add').post((req, res) => {
+router.route('/add').post(async (req, res) => {
 	const username = req.body.username;
-	const firebase_id = req.body.firebase_id;
+	const firebase_id_token = req.body.firebase_id_token;
+	let uid;
+	try {
+		const decodedToken = await admin.auth().verifyIdToken(firebase_id_token);
+		uid = decodedToken.uid;
+	} catch (err) {
+		console.log('Error ' + err);
+		res.status(400).json(err);
+		return;
+	}
```

Comment:
The problem is we need the try-catch block in each different route because if the the token is wrong we need to return the error. So the only thing I would make a helper function is line 24 basically. Do you have any suggestions? (or is there anything I'm missing?)